middleware_accesstoken.go

```diff
@@ -23,19 +25,59 @@ Example usage:
 		})).Methods("POST")
 */
 func NewMiddlewareAccessToken(headerName string, tokens []string) func(rw http.ResponseWriter, req *http.Request) *Response {
+	return newAccessTokenHandler(headerName, tokens, "header")
+}
+
+/*
+NewMiddlewareAccessQueryToken creates a new handler to verify access tokens passed as a query parameter.
+
+Example usage:
+
+	routes.Handle("/some/route", a.Dependencies.MWHandler.Handle(
+		[]rye.Handler{
+			rye.NewMiddlewareAccessQueryToken(queryParamName, []string{token1, token2}),
+			yourHandler,
+		})).Methods("POST")
+*/
+func NewMiddlewareAccessQueryToken(queryParamName string, tokens []string) func(rw http.ResponseWriter, req *http.Request) *Response {
+	return newAccessTokenHandler(queryParamName, tokens, "query")
+}
+
+func newAccessTokenHandler(name string, tokens []string, tokenType string) func(rw http.ResponseWriter, req *http.Request) *Response {
 	a := &accessTokens{
-		headerName: headerName,
-		tokens:     tokens,
+		paramName: name,
+		tokens:    tokens,
+	}
+
+	if tokenType == "query" {
```

Comment:
I think a switch statement would be nice here - it's easier to read and see what's happening (and you can also set the default case to err if the tokenType is not supported (to prevent future, potential errors)).